spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -161,31 +161,7 @@
 
 	private RetryOperations retryOperations;
 
-	/**
-	 * Used when writing directly to ZK. This is what Kafka expects.
-	 */
-	public final static ZkSerializer utf8Serializer = new ZkSerializer() {
-
-		@Override
-		public byte[] serialize(Object data) throws ZkMarshallingError {
-			try {
-				return ((String) data).getBytes("UTF-8");
-			}
-			catch (UnsupportedEncodingException e) {
-				throw new ZkMarshallingError(e);
-			}
-		}
-
-		@Override
-		public Object deserialize(byte[] bytes) throws ZkMarshallingError {
-			try {
-				return new String(bytes, "UTF-8");
-			}
-			catch (UnsupportedEncodingException e) {
-				throw new ZkMarshallingError(e);
-			}
-		}
-	};
+	private Map<String, Collection<Partition>> topicsInUse = new HashMap<>();
```

Comment:
Given binding lifecycle is same for all the bindings in the context, I believe it is ok not removing the `topics` from this map when the `unbind` operation occurs.